packages/tool-cache/src/tool-cache.ts

```diff
@@ -254,6 +255,35 @@ export async function extractTar(
   return dest
 }
 
+/**
+ * Extract a xar compatible archive
+ *
+ * @param file     path to the archive
+ * @param dest     destination directory. Optional.
+ * @param flags    flags for the xar. Optional.
+ * @returns        path to the destination directory
+ */
+export async function extractXar(
+  file: string,
+  dest?: string,
+  flags: string = '-x'
+): Promise<string> {
+  ok(IS_MAC, 'extractXar() not supported on current OS')
+  ok(file, 'parameter "file" is required')
+
+  dest = dest || (await _createExtractFolder(dest))
+  const args = [flags, '-C', dest, '-f', file]
+
+  if (core.isDebug() && !flags.includes('v')) {
+    args.push('-v')
+  }
```

Comment:
considering the function is `extractTar`, we should also push a -x flag if it does not already exist. A unit test for this functionality would be great as well!